### ISSUE TITLE
Improvement to Cape contract

### ISSUE THREAD

**Issue by alxiong** [EspressoSystems/cape#465]
1. `_isBlockEmpty()`: not used, should remove
2. ~`_computeAssetDescription()` is called only once, can be inlined; why need all those “TRICAPE ERC20, sponsored by” in the description? Just directly concatenate the two addresses?~
3. `_checkForeignAssetCode()` should include `address sponsor` in input params to be independent of the caller of this function?
4. `_checkDomesticAssetCode()` can be compress into one line to avoid awkward temp var name “randomBytes”
5. `depositErc20()`: ERC20 -> IERC20 for smaller contract size
6. Another simplification: separate deposit processing out as a single function, ~set a `uint256 constant MIN_DEPOSIT`, in `submitCapeBlock()` check if the `queueSize() > MIN_DEPOSIT`, if so, revert; else proceed to process the block but exclude processing all those deposits. This requires relayers to call `processDeposit()` (or maybe some other names) when there's accumulating deposits in the queue.~ (⚠️ TODO in a separate issue)
7. Add more code comments
8. Use [`safeTransfer`](https://github.com/Uniswap/v3-core/blob/main/contracts/libraries/TransferHelper.sol#L14) from latest Uniswap code or [`SafeTransferLib`](https://github.com/Rari-Capital/solmate/blob/main/src/utils/SafeTransferLib.sol) from solmate. ~Use [`_safeTransfer()` from Uniswap](https://github.com/Uniswap/v2-core/blob/master/contracts/UniswapV2Pair.sol#L44-L47)~ inside `_handleWithdraw()`
9. `Queue.sol` seems a misnomer, it doesn't have `pop()`, nor do we uphold invariant of a proper queue. (e.g. using its current API, the sequence "push item, push item, emptyQueue, getQueueElem(1)" still returns a valid value which violates how a queue should work) -- Instead, we can use a in-storage array which supports `delete` or proper emptying. 

WDYT? 
(cc @sveitser @philippecamacho)
